sql/wsrep_thd.cc

```diff
@@ -328,10 +328,20 @@ void wsrep_fire_rollbacker(THD *thd) {
 int wsrep_abort_thd(const THD *bf_thd, THD *victim_thd, bool signal) {
   DBUG_ENTER("wsrep_abort_thd");
   mysql_mutex_lock(&victim_thd->LOCK_wsrep_thd);
-  if ((WSREP(bf_thd) ||
-       ((WSREP_ON || bf_thd->variables.wsrep_OSU_method == WSREP_OSU_RSU) &&
-        wsrep_thd_is_toi(bf_thd))) &&
-      victim_thd && !wsrep_thd_is_aborting(victim_thd)) {
+
+  // TOI => wsrep_on=1
+  // RSU => wsrep_on=0
+  // The transaction will be aborted if:
+  // (
+  //   1. BF-ing thread has wsrep_on=1
+  //   OR
+  //   2. wsrep is globally enabled and we are in RSU
+  // )
+  // AND
+  // 3. victim thread is not aborting or already aborted
+  if (  (WSREP(bf_thd) ||
+        (WSREP_ON && wsrep_thd_is_in_rsu(bf_thd)))
+      &&  victim_thd && !wsrep_thd_is_aborting(victim_thd) ) {
```

Comment:
Just a thought.

Considering the fact that BF Abort can happen only when bf thread is either
1. An applier thread 
  OR
2. running in TOI
  OR
3. running in RSU

Do you foresee any problem if we write it like this
```
if (
    (   
    ┆ wsrep_thd_is_applying(bf_thd) ||          => BF thread has wsrep_on
    ┆ (WSREP_ON && wsrep_thd_is_rsu(bf_thd)) || => condition for RSU
    ┆ wsrep_thd_is_toi(bf_thd)                  => condition for TOI
    )   
    ┆ && victim_thd 
    ┆ && !wsrep_thd_is_aborting(victim_thd)
   )
```

Reply:
Probably yes, but I would rather stick to the original code condition to keep it as similar to upstream as possible.